raiden/tests/integration/long_running/flaky/test_token_networks.py

```diff
@@ -231,4 +219,5 @@ def test_participant_selection(raiden_network, token_addresses, skip_if_tester):
         token_address=token_address,
         partner_address=sender.address,
     )
-    assert receiver_channel[0].settle_transaction is not None
+    # because of timing, channel may already have been cleaned
```

Comment:
You can use `waiting.wait_for_settle` here and then the settle transaction must be set.

Btw, the channel will not be cleared after settle, but after the unlocks.